Userland/Libraries/LibAudio/MP3Loader.cpp

```diff
@@ -144,7 +228,682 @@ bool MP3LoaderPlugin::synchronize()
+        band_index_ms_end = band_index_intensity_start;
+    }
+
+    for (size_t band_index = band_index_ms_start; band_index < band_index_ms_end; band_index++) {
+        process_ms_stereo(scale_factor_bands[band_index]);
+    }
+
+    for (size_t band_index = band_index_intensity_start; band_index < band_index_intensity_end; band_index++) {
+        int const is_pos = frame.channels[1].scale_factors[band_index];
+        if (is_pos == 7) {
+            if (has_flag(frame.header.mode_extension, MP3::ModeExtension::MsStereo))
+                process_ms_stereo(scale_factor_bands[band_index]);
+            continue;
+        }
+        double const is_ratio = AK::tan(is_pos * AK::Pi<double> / 12);
+        process_intensity_stereo(scale_factor_bands[band_index], is_ratio);
+    }
+}
+
+void MP3LoaderPlugin::transform_samples_to_time(Array<double, 576> const& input, size_t input_offset, Array<double, 36>& output, MP3::BlockType block_type)
+{
+    if (block_type == MP3::BlockType::Short) {
+        int const N = 12;
+        Array<double, N * 3> temp_out;
+        Array<double, N / 2> temp_in;
+
+        for (size_t k = 0; k < N / 2; k++)
+            temp_in[k] = input[input_offset + 3 * k + 0];
+        s_mdct_12.inverse(temp_in, Span<double>(temp_out).slice(0, N));
+        for (size_t i = 0; i < N; i++)
+            temp_out[i + 0] *= MP3::Tables::WindowBlockType2[i];
+
+        for (size_t k = 0; k < N / 2; k++)
+            temp_in[k] = input[input_offset + 3 * k + 1];
+        s_mdct_12.inverse(temp_in, Span<double>(temp_out).slice(12, N));
+        for (size_t i = 0; i < N; i++)
+            temp_out[i + 12] *= MP3::Tables::WindowBlockType2[i];
+
+        for (size_t k = 0; k < N / 2; k++)
+            temp_in[k] = input[input_offset + 3 * k + 2];
+        s_mdct_12.inverse(temp_in, Span<double>(temp_out).slice(24, N));
+        for (size_t i = 0; i < N; i++)
+            temp_out[i + 24] *= MP3::Tables::WindowBlockType2[i];
+
+        Span<double> idmct1 = Span<double>(temp_out).slice(0, 12);
+        Span<double> idmct2 = Span<double>(temp_out).slice(12, 12);
+        Span<double> idmct3 = Span<double>(temp_out).slice(24, 12);
+        for (size_t i = 0; i < 6; i++)
+            output[i] = 0;
+        for (size_t i = 6; i < 12; i++)
+            output[i] = idmct1[i - 6];
+        for (size_t i = 12; i < 18; i++)
+            output[i] = idmct1[i - 6] + idmct2[i - 12];
+        for (size_t i = 18; i < 24; i++)
+            output[i] = idmct2[i - 12] + idmct3[i - 18];
+        for (size_t i = 24; i < 30; i++)
+            output[i] = idmct3[i - 18];
+        for (size_t i = 30; i < 36; i++)
+            output[i] = 0;
+
+    } else {
+        s_mdct_36.inverse(Span<double const>(input).slice(input_offset, 18), output);
+        for (size_t i = 0; i < 36; i++) {
+            switch (block_type) {
+            case MP3::BlockType::Normal:
+                output[i] *= MP3::Tables::WindowBlockType0[i];
```

Comment:
Why are the WindowBlockType tables named 0,1,2,3 if they're only used for the normal, start, short and end blocks respectively? Or is that an ISO/IEC name?

Reply:
It's confusing at times. They like to switch it up. Anyways, renamed.